activeBranch/state.moving.js

```diff
@@ -0,0 +1,38 @@
+const State = require("state.base");
+const { StateTypes } = require("state.base");
+require("protoMod.creep")
+class MovingState extends State {
```

Comment:
Is it worth it to even have a moving state? It's really not hard to incorporate move code into sourcing and working. It's one line, plus a single conditional to execute action. Working makes it easier, as the move code can be in the state as it's consistent across, and then action in the strategy can report being out of range. Seems to be worthwhile.

Reply:
Follow up: maybe keep moving state as a generic move-creep-from-place-to-place" option. Likely going to go primarily unused. 

Go ahead and write the rest of the states as proposed, adding in movement. Should make things easier to follow for the time being. Maybe going to regret that? Hopefully not.